bigchaindb_driver/offchain.py

```diff
@@ -0,0 +1,229 @@
+
+    Raises:
+        :class:`~.exceptions.BigchaindbException`: If ``operation`` is
+            not ``'CREATE'`` or ``'TRANSFER'``.
+
+    .. important::
+
+        * For ``operation='CREATE'``, ``owners_before`` MUST be set,
+          meanwhile ``owners_after``, ``asset``, amd ``metadata`` MAY be
+          set. The argument ``inputs`` is ignored.
+
+        * For ``operation='TRANSFER'``, ``owners_after``, ``asset``,
+          and ``inputs`` MUST be set, meanwhile  ``mwtadata`` MAY be set.
+          The argument ``owners_before`` is ignored.
+
+    """
+    operation = _normalize_operation(operation)
+    return _prepare_transaction(
+        operation,
+        owners_before=owners_before,
+        owners_after=owners_after,
+        asset=asset,
+        metadata=metadata,
+        inputs=inputs,
+    )
+
+
+def prepare_create_transaction(*,
+                               owners_before,
+                               owners_after=None,
+                               asset=None,
+                               metadata=None):
+    """
+    Prepares a ``"CREATE"`` transaction payload, ready to be
+    fulfilled.
+
+    Args:
+        owners_before (:obj:`list` | :obj:`tuple` | :obj:`str`): One
+            or more public keys representing the original owner(s) of
+            the asset being created.
+        owners_after (:obj:`list` | :obj:`tuple` | :obj:`str`, optional):
+            One or more public keys representing the new owner(s) of the
+            asset being created. Defaults to ``None``.
+        asset (:obj:`dict`, optional): The asset being created. Defaults
+            to ``None``.
+        metadata (:obj:`dict`, optional): Metadata associated with the
+            transaction. Defaults to ``None``.
+
+    Returns:
+        dict: The prepared ``"CREATE"`` transaction.
+
+    """
+    if not isinstance(owners_before, (list, tuple)):
+        owners_before = [owners_before]
+
+    # NOTE: Needed for the time being. See
+    # https://github.com/bigchaindb/bigchaindb/issues/797
+    if isinstance(owners_before, tuple):
+        owners_before = list(owners_before)
+
+    if not owners_after:
+        owners_after = owners_before
+
+    asset = _normalize_asset(asset)
+
+    transaction = Transaction.create(
+        owners_before,
+        owners_after,
+        metadata=metadata,
+        asset=asset,
+    )
+    return transaction.to_dict()
+
+
+def prepare_transfer_transaction(*,
+                                 inputs,
+                                 owners_after,
+                                 asset,
+                                 metadata=None):
+    """
+    Prepares a ``"TRANSFER"`` transaction payload, ready to be
+    fulfilled.
+
+    Args:
+        inputs (:obj:`dict` | :obj:`list` | :obj:`tuple`): One or more
+            inputs holding the condition(s) that this transaction
+            intents to fulfill. Each input is expected to be a
+            :obj:`dict`.
```

Comment:
I assume you are suggesting it could be documented? That would be a good point.

It is the complete outer structure. Say you have a `tx` payload you are using as an "input":

```python
cid = 0
condition = tx['transaction']['conditions'][cid]
{
    'fulfillment': condition['condition']['details'],
    'input': {
        'cid': cid,
        'txid': tx['id'],
    },
    'owners_before': condition['owners_after'],
}
```